### ISSUE TITLE
Wygaszamy WebAPI do obsługi ofert i sprzedaży. Korzystaj z REST API.

### ISSUE THREAD

**Issue by TomekKaczmarek** [allegro/allegro-api#277]
**Rozwijamy REST API**
Do końca czerwca 2018 zasoby, które pozwalają [wystawiać i edytować oferty](https://developer.allegro.pl/sale/), są w wersji beta. Jesteśmy otwarci i chętnie wysłuchamy twoich uwag, sugestii i komentarzy. Dlatego tym bardziej zachęcamy do testów. 

**Harmonogram dla REST API**

- **Lipiec 2018** - zasoby, które pozwalają wystawiać i edytować oferty udostępnimy w wersji public.
- **Lipiec 2018** - udostępnimy zasoby do pobierania i zarządzania zamówieniami z Allegro w wersji beta.
- **Październik 2018** - udostępnimy zasoby dla zamówień w wersji public

Docelowo REST API Allegro będzie jedynym interfejsem aplikacyjnym, który pozwoli ci zarządzać ofertą i zamówieniami.

**Wygaszamy WebAPI**

- Nowe funkcjonalności będziemy udostępniać tylko w REST API.
- W **październiku 2018** oznaczymy metody do obsługi oferty i zamówień jako [deprecated](https://en.wikipedia.org/wiki/Deprecation#Software_deprecation). Od tego momentu, będziemy odradzać użycia tych metod w nowych rozwiązaniach i mocno zalecać migrację na REST API w istniejących rozwiązaniach.
- W **kwietniu 2019** usuwamy metody do obsługi oferty i zamówień.

Więcej informacji na ten temat znajdziecie w [tym artykule](https://developer.allegro.pl/news/2018-06-06-Wygaszamy_webapi/).

**Comment by jakubmaguza** [comment]
Mam jedno pytanie: Jak zamierzacie rozwiązać opcję logowania się do RESTAPI dla aplikacji desktopowych?
Z tego co wiem, w tym momencie konieczne jest posiadanie strony internetowej, na którą wyślecie klucz uwierzytelniający. Aktualnie, używając WebApi, mogę zalogować się podając login i hasło. W zamian dostaję klucz sesji.  Jak mogę przeprowadzić takie logowanie przy użyciu REST-API nie mając strony internetowej?

**Comment by anih** [comment]
I jeszcze pytanie jak i czy w ogóle zamierzacie pozwolić na tokeny niewygasające lub wygasające po np. kilku latach albo możliwość zdobycia tokena poprzez metodę w której podaje się login i hasło? Odświeżanie co roku jest dla nas trochę niewygodne i trzeba o tym pamiętać.

**Comment by TomekKaczmarek** [comment]
@jakubmaguza Twoja aplikacja desktopowa musi współpracować z przeglądarką internetową. Może mieć przeglądarkę wbudowaną w aplikację lub korzystać z zewnętrznej zainstalowanej na komputerze użytkownika.

Jeśli twoja aplikacja nie ma jeszcze takiej możliwości możesz:

- Zainstalować wtyczkę WebView - ustaw wtyczkę tak, aby korzystała z przeglądarki systemowej zainstalowanej na komputerze użytkownika. Następnie ustaw w swojej aplikacji adres URI do przekierowań, dzięki niemu będzie w stanie otrzymać od nas prośbę o uwierzytelnienie i odebrać kod autoryzacyjny.
- Zainstalować wtyczkę WebView i dodatkowo skonfigurować tzw. Localhost, ustaw na jakim porcie będzie on nasłuchiwał. Adres ten następnie podaj podczas rejestracji aplikacji jako adres URI do przekierowania. Ustawienie portu serwera lokalnego pozwoli tobie otrzymać od nas prośbę o uwierzytelnienie i odebrać kod autoryzacyjny.

**Comment by TomekKaczmarek** [comment]
@anih nad tym tematem pracujemy - szczegóły podamy jak tylko zdecydujemy się na konkretne rozwiązanie.

**Comment by Feveria** [comment]
> Docelowo REST API Allegro będzie jedynym interfejsem aplikacyjnym, który pozwoli ci zarządzać ofertą i zamówieniami.

Czy REST API pozwoli na wyszukiwanie ofert? Z tego co widzę, na razie taka możliwość jest na WebAPI i nie wiem, czy się babrać w SOAPie w Pythonie, czy czekać na implementację, czy może już jest taka możliwość, ale to przeoczyłem.

**Comment by pfunkowski** [comment]
W jaki sposób zautoryzowac aplikację bez gui? Np. Windows service

śr., 6 cze 2018, 14:14 użytkownik Artur Szcześniak <notifications@github.com>
napisał:

**Comment by esprzedaz** [comment]
Macie w planach umożliwić edycję przez API wszystkiego, co jest potrzebne do wystawienia oferty? Czyli warunków, usług dodatkowych, tabel rozmiarowych, danych kontaktowych (do ogłoszeń) i, przede wszystkim, cenników dostaw? Bez tego ostatniego nawet nie ma sensu zabierać się za testy nowych metod.

**Comment by TomekKaczmarek** [comment]
@Feveria Tak, lada chwila opublikujemy listing ofert i wyszukiwarkę w REST API. Nowy zasób zastąpi docelowo metodę doGetItemList. Więcej informacji podamy w osobnym komunikacie.

**Comment by SoftWareMG** [comment]
A kiedy planujecie calkowicie wylaczyc WebAPI, jest jakas data? Wysłane ze smartfona w T-Mobile, sieci największych możliwości.-------- Wiadomość oryginalna --------Temat: Re: [allegro/allegro-api] Wygaszamy WebAPI do obsługi ofert i sprzedaży. Korzystaj z REST API. (#277)Od: esprzedaz Do: allegro/allegro-api DW: Subscribed Macie w planach umożliwić edycję przez API wszystkiego, co jest potrzebne do wystawienia oferty? Czyli warunków, usług dodatkowych, tabel rozmiarowych, danych kontaktowych (do ogłoszeń) i, przede wszystkim, cenników dostaw? Bez tego ostatniego nawet nie ma sensu zabierać się za testy nowych metod.

—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or mute the thread.
{"@context":"http://schema.org","@type":"EmailMessage","potentialAction":{"@type":"ViewAction","target":"https://github.com/allegro/allegro-api/issues/277#issuecomment-395063016","url":"https://github.com/allegro/allegro-api/issues/277#issuecomment-395063016","name":"View Issue"},"description":"View this Issue on GitHub","publisher":{"@type":"Organization","name":"GitHub","url":"https://github.com"}}
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/allegro/allegro-api","title":"allegro/allegro-api","subtitle":"GitHub repository","main_image_url":"https://assets-cdn.github.com/images/email/message_cards/header.png","avatar_image_url":"https://assets-cdn.github.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/allegro/allegro-api"}},"updates":{"snippets":[{"icon":"PERSON","message":"@esprzedaz in #277: Macie w planach umożliwić edycję przez API wszystkiego, co jest potrzebne do wystawienia oferty? Czyli warunków, usług dodatkowych, tabel rozmiarowych, danych kontaktowych (do ogłoszeń) i, przede wszystkim, cenników dostaw? Bez tego ostatniego nawet nie ma sensu zabierać się za testy nowych metod."}],"action":{"name":"View Issue","url":"https://github.com/allegro/allegro-api/issues/277#issuecomment-395063016"}}}
{
"@type": "MessageCard",
"@context": "http://schema.org/extensions",
"hideOriginalBody": "false",
"originator": "37567f93-e2a7-4e2a-ad37-a9160fc62647",
"title": "Re: [allegro/allegro-api] Wygaszamy WebAPI do obsługi ofert i sprzedaży. Korzystaj z REST API. (#277)",
"sections": [
{
"text": "",
"activityTitle": "**esprzedaz**",
"activityImage": "https://assets-cdn.github.com/images/email/message_cards/avatar.png",
"activitySubtitle": "@esprzedaz",
"facts": [

]
}
],
"potentialAction": [
{
"name": "Add a comment",
"@type": "ActionCard",
"inputs": [
{
"isMultiLine": true,
"@type": "TextInput",
"id": "IssueComment",
"isRequired": false
}
],
"actions": [
{
"name": "Comment",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueComment\",\n\"repositoryFullName\": \"allegro/allegro-api\",\n\"issueId\": 277,\n\"IssueComment\": \"{{IssueComment.value}}\"\n}"
}
]
},
{
"name": "Close issue",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueClose\",\n\"repositoryFullName\": \"allegro/allegro-api\",\n\"issueId\": 277\n}"
},
{
"targets": [
{
"os": "default",
"uri": "https://github.com/allegro/allegro-api/issues/277#issuecomment-395063016"
}
],
"@type": "OpenUri",
"name": "View on GitHub"
},
{
"name": "Unsubscribe",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"MuteNotification\",\n\"threadId\": 342942972\n}"
}
],
"themeColor": "26292E"
}

**Comment by TomekKaczmarek** [comment]
@pfunkowski obecnie nie ma takiej możliwości. Oauth wymaga autoryzacji użytkownika i zezwolenia na dostęp do jego danych. Być może damy taką opcje dla metod, które korzystają z danych publicznych.

**Comment by TomekKaczmarek** [comment]
@esprzedaz 

- Kontakty są: https://developer.allegro.pl/documentation/#/offer-contacts
- Punkty odbioru również: https://developer.allegro.pl/documentation/#/points-of-service
- Tak samo usługi dodatkowe: https://developer.allegro.pl/documentation/#/offer-additional-services
- Cenniki dostaw powinny się wkrótce pojawić
- Tabele rozmiarów również mamy w planach
- Warunki oferty - zdecydowana większość sprzedawców ma po jednym warunku gwarancji, reklamacji i zwrotów - dlatego w najbliższym czasie nie mamy tej opcji w planach. Sprzedawca powinien je ustawić po stronie frontu Allegro. Identyfikatory warunków pobierzesz tymi zasobami: https://developer.allegro.pl/documentation/#/after-sales-service-conditions

**Comment by Feveria** [comment]
@TomekKaczmarek - dzięki za odpowiedź i za implementację nowego API. Dobra robota dla całego zespołu :)

**Comment by aktywnitu** [comment]
Czy developerzy mogliby się dowiedzieć w, którym miesiącu metody do pobierania zamówień (order-management) zostaną włączone na produkcji?

**Comment by PawelTaberski** [comment]
@aktywnitu Nie bardzo rozumiem Twoje pytanie odpowiednie metody do zarządzania zamówieniami są już dostępne w REST API na produkcji od dłuższego czasu #474 . Nawet na ta potrzebę powstał[ odpowiedni poradnik.](https://developer.allegro.pl/orders/)

**Comment by aktywnitu** [comment]
Zgadza się, wyraziłem się nie jasno. W tej chwili metody te są oznaczone jako [BETA]. Kiedy naprawicie wszystkie błędy związane z tymi metodami i będzie można z nich spokojnie korzystać na produkcji?

**Comment by PawelTaberski** [comment]
@aktywnitu Najprawdopodobniej w pierwszym kwartale tego roku. Jednakże już dziś można z nich bezpiecznie korzystać są w wersji beta bo ciągle czekamy na wasze uwagi/sugestie.

**Comment by r-a-f** [comment]
Więcej informacji na ten temat znajdziecie w **tym artykule.**
link nieaktywny!

**Comment by PawelTaberski** [comment]
@r-a-f  Proszę Cię bardzo nowy [link](https://developer.allegro.pl/news/2018-06-06-wygaszamy_webapi/). Jednakże te informacje są już przeterminowane aktualne znajdziesz tutaj #1043
